mula/scheduler/connectors/services/katalogus.py

```diff
@@ -161,74 +164,86 @@ def get_organisations(self) -> list[Organisation]:
+            return _get_from_cache()
         except dict_utils.ExpiredError:
-            self.flush_organisations_boefje_type_cache()
-            return dict_utils.deep_get(self.organisations_boefje_type_cache, [organisation_id, boefje_type])
+            self.flush_boefje_cache()
+            return _get_from_cache()
 
     def get_normalizers_by_org_id_and_type(self, organisation_id: str, normalizer_type: str) -> list[Plugin]:
-        try:
-            with self.lock:
+        def _get_from_cache() -> list[Plugin]:
+            with self.normalizer_cache_lock:
                 return dict_utils.deep_get(
-                    self.organisations_normalizer_type_cache,
+                    self.normalizer_cache,
                     [organisation_id, normalizer_type],
                 )
+
+        try:
+            return _get_from_cache()
         except dict_utils.ExpiredError:
-            self.flush_organisations_normalizer_type_cache()
-            return dict_utils.deep_get(
-                self.organisations_normalizer_type_cache,
-                [organisation_id, normalizer_type],
-            )
+            self.flush_normalizer_cache()
+            return _get_from_cache()
 
     def get_new_boefjes_by_org_id(self, organisation_id: str) -> list[Plugin]:
-        # Get the enabled boefjes for the organisation from katalogus
-        plugins = self.get_plugins_by_organisation(organisation_id)
-        enabled_boefjes = {
-            plugin.id: plugin
-            for plugin in plugins
-            if plugin.enabled is True and plugin.type == "boefje" and plugin.consumes
-        }
-
-        # Check if there are new boefjes
-        new_boefjes = []
-        for boefje_id, boefje in enabled_boefjes.items():
-            if boefje_id in self.organisations_new_boefjes_cache.get(organisation_id, {}):
-                continue
-
-            new_boefjes.append(boefje)
-
-        self.organisations_new_boefjes_cache[organisation_id] = enabled_boefjes
-
-        self.logger.debug(
-            "%d new boefjes found for organisation %s",
-            len(new_boefjes),
-            organisation_id,
-            organisation_id=organisation_id,
-            boefjes=[boefje.name for boefje in new_boefjes],
-        )
-
-        return new_boefjes
+        with self.new_boefjes_cache_lock:
+            # Get the enabled boefjes for the organisation from katalogus
+            plugins = self.get_plugins_by_organisation(organisation_id)
+            enabled_boefjes = {
+                plugin.id: plugin
+                for plugin in plugins
+                if plugin.enabled is True and plugin.type == "boefje" and plugin.consumes
+            }
+
+            # Check if there are new boefjes
+            new_boefjes = []
+            for boefje_id, boefje in enabled_boefjes.items():
+                if boefje_id in self.new_boefjes_cache.get(organisation_id, {}):
+                    continue
+
+                new_boefjes.append(boefje)
```

Comment:
If you use flip the condition by using a `not in` operator, you'll lose the extra else-clause, which makes this imho a bit more maintainable and cleaner

Reply:
yes love it